app/providers/genericGitHubDataService.ts

```diff
@@ -114,16 +169,31 @@ export class GenericGitHubDataService {
     // initiates getting the file content from a repository Url.
     public initiateGetFileContentInPath(repositoryApiUrl: string): Promise<string> {
 
-        console.log('initiating get file content: ' + repositoryApiUrl);
+        // Check if there is an access token saved in local storage
+        var promise = this.local.get("github_access_token").then(github_access_token => {
 
-        return this.http.get(repositoryApiUrl)
-            .toPromise()
-            .then((res) => {
-                return this.handleGetFileContentInPath(res);
-            }).catch((err) => {
-                console.log(JSON.stringify(err));
-                return Promise.reject<string>(err);
-            });
+            // configure headers to include the github credential if the user is signed in
+            let getHeaders = new Headers();
+            let getOptions = new RequestOptions({ headers: getHeaders });
+            if (!!github_access_token) {
+                getOptions.headers.append("Authorization", "Bearer " + github_access_token);
+            }
```

Comment:
You may move this boilerplate code to a helper routine that just returns the RequestOptions, along with token in the headers, as applicable.